### PR TITLE
[CAPPL-324] Fix panic when fetching binary

### PR DESCRIPTION
**Pull request by cedric-cordenier**
* Fix panic when fetching binary
* Lazy load the connector by creating a FetcherService, and starting it after the GatewayConnector starts. This avoids GatewayConnector being nil and prevents us from panicking when calling Fetch.
* Add some logging
* Remove logging of the request/response body -- this was causing my local node to crash since the body was so large.

### CODE REVIEW

**Review by ettec — DISMISSED** [review]

File: core/capabilities/webapi/outgoing_connector_handler.go [hunk 1]
```diff
@@ -125,16 +127,28 @@ func (c *OutgoingConnectorHandler) HandleGatewayMessage(ctx context.Context, gat
...
 
 func (c *OutgoingConnectorHandler) Close() error {
-	return nil
+	return c.StopOnce("OutgoingConnectorHandler", func() error {
+		return nil
+	})
+}
+
+func (c *OutgoingConnectorHandler) HealthReport() map[string]error {
+	return map[string]error{c.Name(): nil}
```
Comment (jmank88) [inline core/capabilities/webapi/outgoing_connector_handler.go]: `StateMachine` implements `Healthy()`:
```suggestion
	return map[string]error{c.Name(): c.Healthy()}
```

File: core/services/workflows/syncer/fetcher.go [hunk 2]
```diff
@@ -2,41 +2,111 @@ package syncer
...
 		}
 
-		return payload.Body, nil
+		s.och = och
+		return och.Start(ctx)
+	})
+}
+
+func (s *FetcherService) Close() error {
+	return nil
```
Comment (jmank88) [inline core/services/workflows/syncer/fetcher.go]: Why not call `s.StopOnce` for completeness?

Also, does `s.och` need to be closed?

File: core/services/workflows/syncer/fetcher.go [hunk 3]
```diff
@@ -2,41 +2,111 @@ package syncer
...
+}
+
+func (s *FetcherService) HealthReport() map[string]error {
+	return map[string]error{s.Name(): nil}
```
Comment (jmank88) [inline core/services/workflows/syncer/fetcher.go]:
```suggestion
	return map[string]error{s.Name(): s.Healthy()}
```

**Comment by lukaszcl** [comment]
Hey @cedric-cordenier! It looks like our internal flaky test detection tool, Flakeguard, has a bug. We're working on a fix. In the meantime, you can ignore these results.

**Review by jmank88 — APPROVED** [review]

**Review by agparadiso — APPROVED** [review]

**Review by ettec — APPROVED** [review]

**Pull request merged**